lib/HALSim/LineSensors.cpp

```diff
@@ -95,34 +95,52 @@ void LineSensors::calibrate()
 
 int16_t LineSensors::readLine()
 {
-    const uint32_t WEIGHT0      = 0;
-    const uint32_t WEIGHT1      = 1000;
-    const uint32_t WEIGHT2      = 2000;
-    const uint32_t WEIGHT3      = 3000;
-    const uint32_t WEIGHT4      = 4000;
+    uint8_t        idx          = 0;
     uint32_t       estimatedPos = 0;
+    uint32_t       numerator    = 0;
+    uint32_t       denominator  = 0;
+    const uint32_t WEIGHT       = 1000;
+    bool           isOnLine     = false;
 
     (void)getSensorValues();
 
+    while (MAX_SENSORS > idx)

```

Comment:
I would prefer a for ( ...) here as it is a normal sequence. Right now the start id is at function top, and the idx increment by one ad body end.  A for would be logical and easier to read (all in one line).

Reply:
With the while() you can keep the variable definition at top together, without redefine the init value again in a nice way.
Anyway I can use a for() loop as well, but would define the variable with its init value inside the loop (don't like this personally).

Reply:
All good, it is just about different personal preferences then. No need to change.  

PS: I like " for (uint32_t idx = 0; ..." for the reason compiler developers  fancy local  and/or nested scopes for locals. It makes optimizing easier.